include/dsn/dist/replication/replication.types.h

```diff
@@ -469,6 +469,14 @@ namespace dsn { namespace replication {
     {
         int32_t app_id;
         int32_t pidx;
+        inline bool operator < (const global_partition_id& r) const
```

Comment:
users should include <dsn/dist/replication.h> instead of the internal </replication/replication.type.h>. It is always the case that a header file does not include everything, e.t., std::hash<global_parititon_id> required for unordered containers.